dist/api_data.py

```diff
@@ -501,6 +501,11 @@ def __ge__(self, other):
             is not limited to not skewing newest, not favoring leaf pages,
             and modifying the eviction score mechanism.''',
             type='boolean'),
+        Config('realloc_exact', 'false', r'''
+            if true, reallocation of memory will only provide the exact
+            amount requested. This will help with spotting memory allocation
+            issues more easily.''',
+            type='boolean'),
```

Comment:
(Not related to this exact location) There should be a python test added for this API. While there isn't actually anything you can measure in the test, it should do something similar to your first test: turn it on, reconfigure it off and do some operations that should cause a reallocation (maybe add in a printf during debugging that does not get committed to know the realloc call was made).